### PR TITLE
test: template search operation

### PR DESCRIPTION
**Pull request by tpluscode**
The minimal changes which were required to get the templated `search` to work close to the spec

### CODE REVIEW

File: examples/blog/api.ttl [hunk 1]
```diff
@@ -10,7 +10,6 @@
 @prefix schema: <http://schema.org/> .
 
 <Blog> a hydra:Class;
-  hydra:search <blog#search>;
```
Comment (tpluscode) [inline examples/blog/api.ttl:13]: As we discussed `hydra:search` is a link property and belongs to the representations

File: examples/blog/blog.js [hunk 2]
```diff
@@ -19,6 +21,17 @@ async function get (req, res) {
...
+    req.hydra.resource.dataset.add(rdf.quad(
+      req.hydra.term,
+      vocab.hydra.view,
+      url,
+    ))
+    req.hydra.resource.dataset.add(rdf.quad(
+      url,
+      vocab.rdf.type,
+      vocab.hydra.PartialCollectionView,
+    ))
```
Comment (tpluscode) [inline examples/blog/blog.js:34]:
```turtle
<> hydra:view </?from=foo;to=bar> .
</?from=foo;to=bar> a hydra:PartialCollectionView .
```

A client requests `/?from=foo;to=bar`
This is the way to tell the client that it should really be interested in the `<>` resource and not the requested on (with query).

The `rdf:type` triple should be unnecessary but needs a change in Alcaeus

File: examples/blog/store-default/__index.nt [hunk 3]
```diff
@@ -1,3 +1,18 @@
...
+
+_:searchTemplate <http://www.w3.org/ns/hydra/core#mapping> _:fromMapping .
+_:fromMapping <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/hydra/core#IriTemplateMapping> .
+_:fromMapping <http://www.w3.org/ns/hydra/core#variable> "from" .
+_:fromMapping <http://www.w3.org/ns/hydra/core#property> <http://localhost:9000/api/schema/from> .
+
+_:searchTemplate <http://www.w3.org/ns/hydra/core#mapping> _:toMapping .
+_:toMapping <http://www.w3.org/ns/hydra/core#property> <http://localhost:9000/api/schema/to> .
+_:toMapping <http://www.w3.org/ns/hydra/core#variable> "to" .
+_:toMapping <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://www.w3.org/ns/hydra/core#IriTemplateMapping> .
```
Comment (tpluscode) [inline examples/blog/store-default/__index.nt:18]: I think I would generate this dynamically in the handler, adding to `req.hydra.resource.dataset`.

On the other hand we eventually want to have this loaded from the API Documentation on the client but that is not implemented.

What do you think?

Comment (bergos) [reply examples/blog/store-default/__index.nt:18]: We can add the code to generate it later.

**Pull request merged**